src/sagemaker/workflow/_utils.py

```diff
@@ -493,8 +496,13 @@ def arguments(self) -> RequestType:
             request_dict.pop("Description")
             logger.warning(warn_msg_template, "Description")
         if "ModelPackageName" in request_dict:
-            request_dict.pop("ModelPackageName")
-            logger.warning(warn_msg_template, "ModelPackageName")
+            if not _pipeline_config or not _pipeline_config.use_custom_job_prefix:
+                request_dict.pop("ModelPackageName", None)
+                logger.warning(warn_msg_template, "ModelPackageName")
```

Comment:
Can you edit this warn_msg_template to include information regarding the new `pipeline_definition_config` field?